trio/_core/_thread_cache.py

```diff
@@ -185,10 +195,16 @@ def _work(self):
 
 
 class ThreadCache:
-    def __init__(self):
-        self._idle_workers = {}
-
-    def start_thread_soon(self, fn, deliver, name: Optional[str] = None):
+    def __init__(self) -> None:
+        self._idle_workers: dict[WorkerThread[Any], None] = {}
```

Comment:
I'm thinking `[object]` works here? Maybe? I'm can't see where we add stuff into this dictionary...

Reply:
It is actually type-unsafe. The problem is in `ThreadCache.start_thread_soon()` - when we pop off an existing worker thread, that effectively is changing the type of the generic `T` from whatever the worker was before. Maybe I should make it `object` here so I can add a type-ignore there to highlight that.

Reply:
Ahh duh, I see.

Reply:
Okay yeah I was thinking maybe there was some covariant coolness we could do because we don't really care about the T used in one function, so maybe it could be an object.

I think I see how that's not really possible here and IMO Any is fine then. Thanks for explaining.

Reply:
I originally had `WorkerThread` non-generic, but that just moves the cast into the `worker._job =` assignment. Which is a bit less safe since you're simultaneously ignoring the type of both callables.